core/shared/src/main/scala/zio/prelude/newtypes/package.scala

```diff
@@ -107,4 +107,51 @@ package object newtypes {
   object FailureOut extends NewtypeF
 
   type FailureOut[+A] = FailureOut.Type[A]
+
+  object Natural extends SubtypeSmart[Int](isGreaterThanEqualTo(0)) {
+
+    val one: Natural =
+      Natural(1)
+
+    val zero: Natural =
+      Natural(0)
+
+    def successor(n: Natural): Natural =
+      Natural(n + 1)
+
+    /**
+     * The `Commutative` and `Identity` instance for the product of `Natural` values.
+     */
+    implicit val NaturalProdCommutativeIdentity: Commutative[Prod[Natural]] with Identity[Prod[Natural]] =
```

Comment:
Since `Nat` is not a class/trait, it can't have a companion object and so we have to put Type class implementations to the companion objects of the Type classes themselves.

Reply:
Yes. I thought we had that functionality with our new types but looks like not yet. I moved it now.